tests/checkpoint/test_checkpoint_result_format.py

```diff
@@ -0,0 +1,620 @@
+import logging
+from typing import List
+
+import pytest
+
+from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
+from great_expectations.core import (
+    ExpectationConfiguration,
+    ExpectationSuiteValidationResult,
+)
+from great_expectations.core.yaml_handler import YAMLHandler
+from great_expectations.data_context.data_context.data_context import DataContext
+from great_expectations.data_context.types.base import CheckpointConfig
+from great_expectations.util import filter_properties_dict
+
+yaml = YAMLHandler()
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def reference_checkpoint_config_for_unexpected_column_names() -> dict:
+    """
+    This is a reference checkpoint dict. It is not used by the tests on its own but subsequent functions will add
+    runtime_configurations that will then be tested.
+
+    checkpoint_dict_unexpected_index_column_names_defined_one_column()
+        - adds runtime_configuration where "unexpected_index_column_names": ["pk_1"],
+    checkpoint_dict_unexpected_index_column_names_defined_two_columns()
+        - adds runtime_configuration where "unexpected_index_column_names": ["pk_1", "pk_2"],
+    checkpoint_dict_unexpected_index_column_names_not_defined()
+        - adds runtime_configuration where "unexpected_index_column_names" are defined
```

Comment:
Love the comprehensive test cases and helpful docstrings. Two nits - I think this line should be `are not defined` and these tests should be marked integration since they spin up a full data context (incl config on filesystem) and associated GX objects.